lightwood/model/neural.py

```diff
@@ -197,8 +199,8 @@ def _max_fit(self, train_dl, dev_dl, criterion, optimizer, scaler, stop_after, r
                 epochs_to_best = epoch
 
             if len(running_errors) >= 5:
-                delta_mean = np.mean([running_errors[-i - 1] - running_errors[-i]
-                                     for i in range(1, len(running_errors[-5:]))])
+                delta_mean = np.average([running_errors[-i - 1] - running_errors[-i] for i in range(1, 5)],
+                                        weights=[(1 / 2)**i for i in range(1, 5)])
```

Comment:
As a note: while this + warmup removal helped, I think it would be nice to compare Neural-only models against results from the latest Native version.

Reply:
I disagree. If it improves benchmark performance overall then it's good to go, if no then it isn't.

Thinking about it this way:
- There's something the neural mixer is good at and some things it isn't
- If it ends up being bad at everything when compared to e.g. lightgbm that's bad and we should fix it or drop it
- If it's good at say 20-50% of things that's fine, and we should focus on those because the other mixers cover the other cases (and this we can track on global benchmarks)
- If we think it can be good at a problem space where it isn't already, this can also be tracked via the global benchmark performance, because "good" should translate to "better than lightgbm"
- As long as a change doesn't make it worst for the problem-space where it's good at, it doesn't really matter if performance drops in other problem spaces.

Long-term maybe it should matter, but we're going to be adding so many new mixers that I'd be way too complex, for now, to give this treatment to all of them. And how would we act on the results, even if we did?

What if a change makes the NN worst overall but better on the problems it's already good at? I'd still say the change is good to go. So would having this suite severely alter our behaviour ?

Reply:
To follow up on a discussion we had this week - the base "behavior" of a model should be expected (i.e. neural should be a vanilla DNN). 

Exotic optimizers, training strategies, and other things should be considered in addition (ideally as abstracted classes etc). I am ok with more "advanced" forms of getting our NN to work (hyper parameter tuning and architecture behavior is non-trivial; realistically getting it to generalize on any dataset with a fixed architecture is not going to work) but let's at least ensure the traditional behavior of a model is captured in the naming scheme.

Reply:
As an aside- what rationale inspired the weights for the average mean? I can understand you want to weight "more recent" differences than later ones, but is this a standard practice?

Reply:
TO ADD - from discussing with @paxcema (2021.09.09) we will convert the weights criteria here into an `EarlyStopping` abstraction for modularity